packages/components/radio/src/radio.tsx

```diff
@@ -154,6 +154,7 @@ export const useRadio = <
       "data-active": dataAttr(isActive),
       "data-hover": dataAttr(isHovered),
       "data-checked": dataAttr(checked),
+      "aria-checked": checked,
```

Comment:
@illionillion 

`aria-checked` should be set on elements with the `radio` role, not on icons.

> If a radio button is checked, the radio element has [aria-checked](https://w3c.github.io/aria/#aria-checked) set to true. If it is not checked, it has [aria-checked](https://w3c.github.io/aria/#aria-checked) set to false.
https://www.w3.org/WAI/ARIA/apg/patterns/radio/

![review-radio-group](https://github.com/yamada-ui/yamada-ui/assets/33865215/f4a9b42e-8cbc-4bbc-a5c9-faa1cd9faa49)

Reply:
I fixed it.
![image](https://github.com/yamada-ui/yamada-ui/assets/60034520/4158bf02-283d-4efa-83e6-9768681ac34e)